src/lib/repeat.ts

```diff
@@ -74,7 +73,7 @@ export function repeat<T>(
       // Try to reuse a part, either keyed or from the list of previous parts
       // if there's no keyMap
       let itemPart =
-          keyMap === undefined ? oldParts[oldPartsIndex++] : keyMap.get(key);
+          keyMap === undefined ? oldParts[0] : keyMap.get(key);
```

Comment:
I see how this fixes the bug, but I don't like the accounting code here, especially the splice + indexOf below. Seems like we shouldn't need both an array and map of old parts. If we have a map, I'd hope we don't need to search through list to find the part there. Hoping we can clean this up a lot :)

Reply:
👍 I'm assuming that most of this will be refactored out in a future change.